content/blog/2019/egress-traffic-control-in-istio-part-1/index.md

```diff
@@ -0,0 +1,156 @@
+I will describe the implementation of the egress traffic control in Istio and compare it with other solutions in the
+next posts.
+
+The most important security aspect for a service mesh is probably ingress traffic. You definitely must prevent attackers
+from penetrating the cluster though ingress APIs. Having said that, securing
+the traffic leaving the mesh is also very important. Once your cluster is compromised, and you must be
+prepared for that scenario, you want to reduce the damage as much as possible and prevent the attackers from using the
+cluster for further attacks on external services and legacy systems outside of the cluster. For that you need egress
+traffic control.
+
+Compliance requirements is another reason to control egress traffic. For example, the [Payment Card
+Industry (PCI) Data Security Standard](https://www.pcisecuritystandards.org/pci_security/) requires that inbound
+and outbound traffic must be restricted to that which is necessary:
+
+{{< quote >}}
+_1.2.1 Restrict inbound and outbound traffic to that which is necessary for the cardholder data environment, and specifically deny all other traffic._
+{{< /quote >}}
+
+Let's start with the attacks that involve egress traffic.
+
+## The attacks
+
+Any IT organization must operate under the assumption that it will be attacked one day, if it is not attacked already, and
+An IT organization must assume it will be attacked if it hasn't been attacked already, and that
+part of its infrastructure could already be compromised or become compromised in the future.
+Once attackers are able to penetrate an application in a cluster, they can proceed to attack external services:
+legacy systems, external web services and databases. The attackers may want to steal the data of the application and to
+transfer it to their external servers. Attackers' malware may require access to attackers' servers to download
+updates. The attackers may use pods in the cluster to perform DDOS attacks or to break into external systems.
+Even though you [cannot know](https://en.wikipedia.org/wiki/There_are_known_knowns) all the possible types of
+attacks, you want to reduce possibilities for any attacks, both for known and unknown ones.
+
+The attackers can be external, that is they can gain access to the application’s container from the outside, through a
+bug in the application. The attackers can also be internal, that is, for example, malicious DevOps people inside the
+organization.
+
+To prevent the attacks described above, some form of egress traffic control must be applied. Let me present egress
+traffic control in the following section.
+
+## The solution: egress traffic control
+
+To prevent attacks involving egress traffic, you must monitor all the egress traffic and enforce all security policies.
+Once you monitor the egress traffic, you can  analyze it, possibly offline, and detect the attacks, even if
+you were unable to prevent them in real time.
```

Comment:
Note that you can prevent attacks by various means, not only by security policies. Policies do not prevent attacks, they reduce possibilities for attacks.

Reply:
Thanks for clarifying. Readers might find that fact helpful too. I mentioned the security policies to help readers make the connection between attack prevention and policies. Consider:
```suggestion
you were unable to prevent them in real time through various means, for example through security policies.
```